### DISCUSSION TITLE
2024 Journey begins: Aryan Nanda

### DISCUSSION THREAD

**Discussion by AryanNanda17** [JdeRobot/jderobot.github.io#239] [General]
Hello everyone, I'm Aryan Nanda, a second-year Computer Science student studying in Mumbai. My background includes working with image processing and various machine-learning models. I'm keen on contributing to this organization, particularly in the realm of deep learning applied to robotics.

Although I have experience with image processing and deep learning, I'm currently acquainting myself with ROS and Docker, which are relatively new to me. I've already started learning about ROS and the workings of Docker.

I would greatly appreciate guidance on how to proceed further and where I could begin making contributions to the organization. I'm enthusiastic about collaborating with the community and eager to start working together!

**Comment by jmplaza** [comment]
Hi @AryanNanda17 , for GSoC just use[ JdeRobot Discussions](https://github.com/orgs/JdeRobot/discussions), instead of [RoboticsAcademy Discussions](https://github.com/JdeRobot/RoboticsAcademy/discussions). RoboticsAcademy is just one of the projects of our JdeRobot organization, maybe the main one, but there are also other projects such as VisualCircuit, FPGA-Robotics, BehaviorMetrics... 

I transfer your message to JdeRobot Discussions

**Reply by AryanNanda17** [reply]
Thank you for providing clarification regarding the discussion platforms.

**Comment by jmplaza** [comment]
Take a look at 
https://github.com/orgs/JdeRobot/discussions/233#discussioncomment-7983733
https://github.com/orgs/JdeRobot/discussions/233#discussioncomment-7983227

**Comment by AryanNanda17** [comment]
@jmplaza
I am a user, interested in the jdeRobot projects on an ARM architecture system(M1 mac). I noticed that the current Docker images appears to be built for x86_64(amd64). Could you please let me know if there is an ARM-compatible Docker image available or if there are plans to provide support for ARM users?

**Reply by jmplaza** [reply]
Hi @AryanNanda17, 

no AFAIK, but maybe you can build your own mini-RADI locally from the source code as long as the dependencies are also available for ARM architectures.

**Comment by AryanNanda17** [comment]
Hello @jmplaza,

I followed your recommendation and successfully set up the mini RADI locally. However, I encountered an issue where the default image platform is Linux/amd64, and I'm unsure why this occurred.

Additionally, I attempted to utilize Rosetta Emulation, but it led to high CPU usage, peaking at 800%. This resulted in significant lag, making it impossible to run the exercises smoothly.

I'm seeking further guidance on how to proceed. My goal is to contribute to jdeRobots in GSOC, but the current environment setup challenges are hindering my progress. Despite having a basic understanding of ROS, Docker, and frontend development, I find it difficult to contribute under these circumstances.

I would appreciate any additional suggestions or solutions you can provide.
Thank you.

**Comment by jmplaza** [comment]
Hi @AryanNanda17 ,

take a look at [this webpage](https://forums.docker.com/t/run-x86-intel-and-arm-based-images-on-apple-silicon-m1-macs/117123) with hints to run x86 (Intel) and ARM based images on Apple Silicon (M1) Macs.

[this video](https://www.youtube.com/watch?v=8xYOgmjDX-c) shows an alternative way too.

Hope it helps.

**Comment by AryanNanda17** [comment]
@jmplaza, I tried both the ways but the issue persists. Then, I raised an [issue](https://github.com/docker/for-mac/issues/7145) in docker/for-mac repo and turns out this is happening because of gzServer process which runs [Gazebo](https://gazebosim.org/home) as an emulated process. The member said that there's not much they can do in Docker Desktop for now.

**Comment by AryanNanda17** [comment]
@jmplaza @sergiopaniego, I want to confirm that will there be any ML, RL, or CV project in jdeRobot this year? 
In the last month, I spent most of my time in Jde in the setup of RoboticsAcademy as gzServer was lagging a lot in my system, it is still lagging in my system but I am able to reduce it somehow. 
I am not interested in FrontEnd Development. But, I have past experience in Ml and Ros and I want to contribute in Jde robot through those domains. If you guys haven't planned any such project, then guide me on how to pitch a good project related to the same domains. 
Please Reply. 
Thanks!

**Reply by jmplaza** [reply]
Hi @AryanNanda17,

it is too soon to say anything solid, as the selected organizations will be announced on February 21st according to [GSoC-2024 timeline](https://developers.google.com/open-source/gsoc/timeline). We have open source projects on AI driven Robotics (ML...) and ROS, such as [BehaviorMetrics](https://github.com/JdeRobot/BehaviorMetrics) or [GSoC-2023 Qi's project](https://github.com/TheRoboticsClub/gsoc2023-Meiqi_Zhao) . Feel free to use that software.
